packages/grid/x-data-grid/src/hooks/features/keyboardNavigation/useGridKeyboardNavigation.ts

```diff
@@ -331,7 +331,10 @@ export const useGridKeyboardNavigation = (
       // Get the most recent params because the cell mode may have changed by another listener
       const cellParams = apiRef.current.getCellParams(params.id, params.field);
 
-      if (cellParams.cellMode !== GridCellModes.Edit && isNavigationKey(event.key)) {
+      if (
+        (cellParams.cellMode !== GridCellModes.Edit && isNavigationKey(event.key)) ||
+        event.key === 'Tab'
```

Comment:
```suggestion
        (cellParams.cellMode !== GridCellModes.Edit  && event.key === 'Tab')
```

This would fix failed e2e test

Reply:
> This would fix failed e2e test

but this was the problem `useGridKeyboardNavigation` won't publish the event in the edit mode.

Reply:
Sorry, I forgot to reverse the check.
I meant this:

```diff
- event.key === 'Tab'
+ (cellParams.cellMode === GridCellModes.Edit  && event.key === 'Tab')
```

Reply:
hey @cherniavskii 

I have adapted @m4theushw solution and I think everything is working now.